app/src/main/kotlin/com/wire/android/util/ui/ScreenSettingsUtil.kt

```diff
@@ -47,10 +50,13 @@ private fun Activity.wakeUpDevice() {
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
         setShowWhenLocked(true)
         setTurnScreenOn(true)
+        val keyguardManager = getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+        keyguardManager.requestDismissKeyguard(this, null)
     } else {
         window.addFlags(
-            WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
-                    or WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
+            WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD or
```

Comment:
I still think that `requestDismissKeyguard()` is not needed and `setShowWhenLocked()` would be enough.

Google says :
Use [FLAG_SHOW_WHEN_LOCKED](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_SHOW_WHEN_LOCKED) or [KeyguardManager#requestDismissKeyguard](https://developer.android.com/reference/android/app/KeyguardManager#requestDismissKeyguard(android.app.Activity,%20android.app.KeyguardManager.KeyguardDismissCallback)) instead of `FLAG_DISMISS_KEYGUARD`.
https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_DISMISS_KEYGUARD

and then
Use [R.attr.showWhenLocked](https://developer.android.com/reference/android/R.attr#showWhenLocked) or [Activity.setShowWhenLocked(boolean)](https://developer.android.com/reference/android/app/Activity#setShowWhenLocked(boolean)) instead of `FLAG_SHOW_WHEN_LOCKED`  to prevent an unintentional double life-cycle event.

https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_SHOW_WHEN_LOCKED

Reply:
The same documentation says that:
>If the value set for the Activity attr android.R.attr.turnScreenOn is true, the screen will turn on when the keyguard is dismissed.

But ok, I removed it - it's easy change to bring back if needed, but on Android 9 without dismissing the keygyard, the screen never turns on when I have an incoming call. With dismissing keyguard, it shows but requires a pin/pattern to unlock the screen before answering, which still isn't ideal but at least the screen turns on and shows that there's an incoming call.
That's why I added it. Maybe there isn't a big demand for fixing it for older Android versions, but we should be aware that it doesn't work well on every Android version.